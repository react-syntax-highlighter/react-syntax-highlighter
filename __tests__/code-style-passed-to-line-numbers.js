import React from 'react';
import renderer from 'react-test-renderer';
import SyntaxHighlighter from '../src';

const code = `const woah = fun => fun + 1;
const dude = woah(2) + 3;
function thisIsAFunction() {
  return [1,2,3].map(n => n + 1).filter(n !== 3);
}
console.log('making up fake code is really hard');

function itIs() {
  return 'no seriously really it is';
}
`;

test('SyntaxHighlighter component passes along code style to non-inline line numbers element', () => {
  const tree = renderer
    .create(
      <SyntaxHighlighter
        codeTagProps={{ style: { color: 'blue' } }}
        showLineNumbers={true}
        showInlineLineNumbers={false}
      >
        {code}
      </SyntaxHighlighter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
