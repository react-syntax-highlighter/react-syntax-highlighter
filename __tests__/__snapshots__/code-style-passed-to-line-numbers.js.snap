// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyntaxHighlighter component passes along code style to non-inline line numbers element 1`] = `
<pre
  style={
    Object {
      "background": "#F0F0F0",
      "color": "#444",
      "display": "block",
      "overflowX": "auto",
      "padding": "0.5em",
    }
  }
>
  <code
    style={
      Object {
        "color": "blue",
        "whiteSpace": "pre",
      }
    }
  >
    <code
      style={
        Object {
          "color": "blue",
          "float": "left",
          "paddingRight": "10px",
        }
      }
    >
      <span
        className="react-syntax-highlighter-line-number"
        style={Object {}}
      >
        1

      </span>
      <span
        className="react-syntax-highlighter-line-number"
        style={Object {}}
      >
        2

      </span>
      <span
        className="react-syntax-highlighter-line-number"
        style={Object {}}
      >
        3

      </span>
      <span
        className="react-syntax-highlighter-line-number"
        style={Object {}}
      >
        4

      </span>
      <span
        className="react-syntax-highlighter-line-number"
        style={Object {}}
      >
        5

      </span>
      <span
        className="react-syntax-highlighter-line-number"
        style={Object {}}
      >
        6

      </span>
      <span
        className="react-syntax-highlighter-line-number"
        style={Object {}}
      >
        7

      </span>
      <span
        className="react-syntax-highlighter-line-number"
        style={Object {}}
      >
        8

      </span>
      <span
        className="react-syntax-highlighter-line-number"
        style={Object {}}
      >
        9

      </span>
      <span
        className="react-syntax-highlighter-line-number"
        style={Object {}}
      >
        10

      </span>
    </code>
    <span
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      const
    </span>
    <span
      style={Object {}}
    >
       woah = 
    </span>
    <span
      className="hljs-function hljs-params"
      style={Object {}}
    >
      fun
    </span>
    <span
      className="hljs-function"
      style={Object {}}
    >
       =&gt;
    </span>
    <span
      style={Object {}}
    >
       fun + 
    </span>
    <span
      style={
        Object {
          "color": "#880000",
        }
      }
    >
      1
    </span>
    <span
      style={Object {}}
    >
      ;

    </span>
    <span
      style={Object {}}
    >
      
    </span>
    <span
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      const
    </span>
    <span
      style={Object {}}
    >
       dude = woah(
    </span>
    <span
      style={
        Object {
          "color": "#880000",
        }
      }
    >
      2
    </span>
    <span
      style={Object {}}
    >
      ) + 
    </span>
    <span
      style={
        Object {
          "color": "#880000",
        }
      }
    >
      3
    </span>
    <span
      style={Object {}}
    >
      ;

    </span>
    <span
      style={Object {}}
    >
      
    </span>
    <span
      className="hljs-function"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      function
    </span>
    <span
      className="hljs-function"
      style={Object {}}
    >
       
    </span>
    <span
      className="hljs-function"
      style={
        Object {
          "color": "#880000",
          "fontWeight": "bold",
        }
      }
    >
      thisIsAFunction
    </span>
    <span
      className="hljs-function"
      style={Object {}}
    >
      (
    </span>
    <span
      className="hljs-function"
      style={Object {}}
    >
      ) 
    </span>
    <span
      style={Object {}}
    >
      {

    </span>
    <span
      style={Object {}}
    >
        
    </span>
    <span
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      return
    </span>
    <span
      style={Object {}}
    >
       [
    </span>
    <span
      style={
        Object {
          "color": "#880000",
        }
      }
    >
      1
    </span>
    <span
      style={Object {}}
    >
      ,
    </span>
    <span
      style={
        Object {
          "color": "#880000",
        }
      }
    >
      2
    </span>
    <span
      style={Object {}}
    >
      ,
    </span>
    <span
      style={
        Object {
          "color": "#880000",
        }
      }
    >
      3
    </span>
    <span
      style={Object {}}
    >
      ].map(
    </span>
    <span
      className="hljs-function hljs-params"
      style={Object {}}
    >
      n
    </span>
    <span
      className="hljs-function"
      style={Object {}}
    >
       =&gt;
    </span>
    <span
      style={Object {}}
    >
       n + 
    </span>
    <span
      style={
        Object {
          "color": "#880000",
        }
      }
    >
      1
    </span>
    <span
      style={Object {}}
    >
      ).filter(n !== 
    </span>
    <span
      style={
        Object {
          "color": "#880000",
        }
      }
    >
      3
    </span>
    <span
      style={Object {}}
    >
      );

    </span>
    }

    <span
      style={Object {}}
    >
      
    </span>
    <span
      style={
        Object {
          "color": "#397300",
        }
      }
    >
      console
    </span>
    <span
      style={Object {}}
    >
      .log(
    </span>
    <span
      style={
        Object {
          "color": "#880000",
        }
      }
    >
      'making up fake code is really hard'
    </span>
    <span
      style={Object {}}
    >
      );

    </span>
    

    <span
      style={Object {}}
    >
      
    </span>
    <span
      className="hljs-function"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      function
    </span>
    <span
      className="hljs-function"
      style={Object {}}
    >
       
    </span>
    <span
      className="hljs-function"
      style={
        Object {
          "color": "#880000",
          "fontWeight": "bold",
        }
      }
    >
      itIs
    </span>
    <span
      className="hljs-function"
      style={Object {}}
    >
      (
    </span>
    <span
      className="hljs-function"
      style={Object {}}
    >
      ) 
    </span>
    <span
      style={Object {}}
    >
      {

    </span>
    <span
      style={Object {}}
    >
        
    </span>
    <span
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      return
    </span>
    <span
      style={Object {}}
    >
       
    </span>
    <span
      style={
        Object {
          "color": "#880000",
        }
      }
    >
      'no seriously really it is'
    </span>
    <span
      style={Object {}}
    >
      ;

    </span>
    }

    
  </code>
</pre>
`;
